### PR TITLE
Secret Keeper Application

### PR DESCRIPTION
**Pull request by chenchanglew**
**What this PR does / why we need it**:

Create a new demo application ~ Secret Keeper

Secret Keeper has 5 functions.
InitSecretKeeper:
- This function will initialize the key "AUTH_LIST_KEY" with value ["Alice", "Bob"] and key "SECRET_KEY" with value "DefaultSecret".
- This function should only be called once when the application started.
- Of course a malicious user can call this function to reset the value, but this we will assume this is not what an attacker would want to achieve. RevealSecret:
- This function allow users that in the Authlist ("AUTH_LIST_KEY") able to reveal the value of secret stored under key "SECRET_KEY". LockSecret:
- This function allow users that in the Authlist ("AUTH_LIST_KEY") able to store a new value of secret under key "SECRET_KEY".
- The old value will be replaced. AddUser:
- This function allow users that in the Authlist ("AUTH_LIST_KEY") able to add a new user to the Authlist.
- Then the new user can now perform the following four functions (RevealSecret, LockSecret, AddUser, RemoveUser) RemoveUser:
- This function allow users that in the Authlist ("AUTH_LIST_KEY") able to add remove an existing user off the Authlist.
- Then the removed user can no longer able to perform the following four functions (RevealSecret, LockSecret, AddUser, RemoveUser)

Example using fpc-simple-client:
./fpcclient invoke initSecretKeeper
./fpcclient query revealSecret Alice
./fpcclient invoke lockSecret Bob NewSecret
./fpcclient query revealSecret Alice
./fpcclient invoke removeUser Alice Bob
./fpcclient query revealSecret Alice
./fpcclient query revealSecret Bob  // (will failed) ./fpcclient invoke addUser Alice Bob
./fpcclient query revealSecret Bob // (will success)

**Which issue(s) this PR fixes**:

This is just create a new application for ppc example

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mbrandenburger — COMMENTED** [review]
What about adding a readme file in the chaincode secret-keeper folder with a description on what it does and how to use it.

**Comment by mbrandenburger** [comment]
Thanks @chenchanglew, the added README looks good to me. 

@munapower could you please check this PR? Thanks

File: samples/chaincode/secret-keeper-go/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+- **LockSecret**: Enables authorized users to update the secret value. This action replaces the existing secret.
+
+- **AddUser**: Permits authorized users to add a new user to the authorization list, granting them access to all functions.
+
+- **RemoveUser**: Allows authorized users to remove an existing user from the authorization list, revoking their access.
+
+## Example Usage
+
+To demonstrate Secret Keeper's capabilities, you can use the `fpc-simple-client` as follows:
```
Comment (munapower) [inline samples/chaincode/secret-keeper-go/README.md:21]: it isn't fpc-simple-client, it is simple-cli-go.

Comment (munapower) [reply samples/chaincode/secret-keeper-go/README.md:21]: and a link to the readme should be added.

File: samples/chaincode/secret-keeper-go/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- **RevealSecret**: Allows authorized users to view the currently stored secret.
+
+- **LockSecret**: Enables authorized users to update the secret value. This action replaces the existing secret.
+
+- **AddUser**: Permits authorized users to add a new user to the authorization list, granting them access to all functions.
+
+- **RemoveUser**: Allows authorized users to remove an existing user from the authorization list, revoking their access.
+
+## Example Usage
+
```
Comment (munapower) [inline samples/chaincode/secret-keeper-go/README.md:20]: I would also add that to test it you can deploy the chaincode to the-simple-testing-network and then invoke it with the simple-cli-go.

**Review by munapower — CHANGES_REQUESTED** [review]
Minor changes to readme to make it easier to follow.

File: samples/chaincode/secret-keeper-go/README.md (hunk 2 above)
Comment (chenchanglew) [reply samples/chaincode/secret-keeper-go/README.md:20]: done

File: samples/chaincode/secret-keeper-go/README.md (hunk 1 above)
Comment (chenchanglew) [reply samples/chaincode/secret-keeper-go/README.md]: done

**Comment by chenchanglew** [comment]
All the comments has been resolved

**Review by munapower — APPROVED** [review]

**Pull request merged**